src/AppInstallerCLITests/TestSource.h

```diff
@@ -17,7 +17,8 @@ namespace TestCommon
         using LocIndString = AppInstaller::Utility::LocIndString;
         using MetadataMap = AppInstaller::Repository::IPackageVersion::Metadata;
 
-        TestPackageVersion(const Manifest& manifest, MetadataMap installationMetadata = {});
+        TestPackageVersion(const Manifest& manifest, std::weak_ptr<const ISource> source = {});

```

Comment:
Was there a lifetime bug?

Reply:
There wasn't one. I changed it from shared to weak because of another change I made. I made the test composite source pass a pointer to `this` to the packages it returns and there was some issue with it being shared, but I can't remember exactly what.